_ml-commons-plugin/extensibility/blueprints.md

```diff
@@ -73,7 +73,97 @@ The `action` parameter supports the following options.
+
+## Built-in pre- and post-processing functions
+
+Call the built-in pre- and post-processing functions instead of writing a custom Painless script when connecting to the following text embedding models or your own text embedding models deployed on a remote server (for example, Amazon SageMaker):
+
+- [Pretrained models provided by OpenSearch](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/)
+- [OpenAI remote models](https://platform.openai.com/docs/api-reference/embeddings)
+- [Cohere remote models](https://docs.cohere.com/reference/embed)
+
+OpenSearch provides the following pre- and post-processing functions:
+
+- OpenAI: `connector.pre_process.openai.embedding` and `connector.post_process.openai.embedding`
+- Cohere: `connector.pre_process.cohere.embedding` and `connector.post_process.cohere.embedding`
+- [Default](#default-pre--and-post-processing-functions) (for neural search): `connector.pre_process.default.embedding` and `connector.post_process.default.embedding`
+
+### Default pre- and post-processing functions
+
+The default pre- and post-processing functions translate between the format that the model expects and the format that [neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/) expects. 
+
+#### Example request
+
+The following example request creates a SageMaker text embedding connector and calls the default post-processing function:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Sagemaker text embedding connector",
+  "description": "The connector to Sagemaker",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "access_key": "<REPLACE WITH SAGEMAKER ACCESS KEY>",
+    "secret_key": "<REPLACE WITH SAGEMAKER SECRET KEY>"
+  },
+  "parameters": {
+    "region": "ap-northeast-1",
+    "service_name": "sagemaker"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "sagemaker.ap-northeast-1.amazonaws.com/endpoints/",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "post_process_function": "connector.post_process.default.embedding",
+      "request_body": "${parameters.input}"
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The `request_body` template must be `${parameters.input}``. 
+{: .important}
+
+### Preprocessing function 
+
+The `connector.pre_process.default.embedding` default preprocessing function parses the neural search request and transforms it into the format that the model expects as input.
+
+The model input format must be a list of strings, for example:
```

Comment:
The model must be able to process such predict request input
```
{
    "parameters": {
        "input": ["hello", "world"]
     }
}
```

Reply:
No, this is the input ml-commons expects, not the remote model, for remote model + default preprocessor, the model itself must accept ["aaa", "bbb"] format.

Reply:
Changed to array of strings.